### DISCUSSION TITLE
Audio Examples using STM32F723DISCO and STM32CubeMX

### DISCUSSION THREAD

**Discussion by M3gaFr3ak** [hathach/tinyusb#830] [Q&A]
I created an STM32CubeMX/STM32CubeIDE project and successfully integrated tinyusb into it. I copied the code from the uac2headset example and my MCU also enumerates as a speaker/microphone combo. 

My problem at the moment is:
According to my understanding of your code (especially the audio_task), this project should work like stereomix, looping the speaker output to the microphone input. This isn't happening for me. I get the LED to signal that it's indeed streaming, but I don't get any signal on the microphone acording to the audio controlpanel when outputting sound on the speaker. spk_data_size inside audio_task actually never goes beyond 0.

Is the uac2headset currently working officially?

**Comment by hathach** [comment]
> Is the uac2headset currently working officially?

Have you tried to run the example as it is to see if that work ? you may miss something while moving it to IDE. Also it depends on the mcu as well, you should always mentioned your MCU since each port is different.

**Reply by M3gaFr3ak** [reply]
I didn't use the example builds initially since they wouldn't build due to a make-misconfiguration on my system. I fixed that now.
Results: CDC and MSC examples work properly. Audio examples wont even enumerate, LEDs also don't do anything. My self-ported Audio example did actually enumerate and blink, just not stream audio.

Controller is the STM32F723E (STM32F723E-DISCO  board)

**Reply by hathach** [reply]
Try to run audio_test example. I have tested it recently. F723 has 2 usb ports, you need to run make properly with PORT= and SPEED=. For using IDE, I couldn't help since I don't use it at all.

**Reply by M3gaFr3ak** [reply]
I used `mingw32-make BOARD=stm32f723disco all` which also printed "Using OTG_HS in HighSpeed mode" (which should be correct, this info is also in the board.mk of the board)

EDIT: Using `PORT=0 SPEED=full` makes it "work" at the fullspeed port, but in Device Manager it shows 

```
This device cannot start. (Code 10)

The specified range could not be found in the range list.
```

**Reply by M3gaFr3ak** [reply]
I tried debugging the audio_test program using the highspeed-USB port as the used port. When interrupting the program, it seems to be stuck in the dcd_init routine explaining why it won't enumerate.
```
Program received signal SIGTRAP, Trace/breakpoint trap.
0x0800268e in dcd_init (rhport=0 '\000') at C:/Users/leonl/repos/tinyusb/src/portable/st/synopsys/dcd_synopsys.c:473
473       while ((usb_otg->GRSTCTL & USB_OTG_GRSTCTL_AHBIDL) == 0U) {}
```

**Reply by hathach** [reply]
hmm, probably issue with HS port, it is newly added. There is similar reset issue here https://github.com/hathach/tinyusb/discussions/769 . Though I am not sure how it is fixed. Haven't got time to actually try it again.  

Note: please make sure you compile with make and existing bsp since that is what I am gonna use to reproduce the issue.

**Reply by M3gaFr3ak** [reply]
I compiled using mingw32-make and the latest pull of the repos master branch.
Is there something I can do to help?

**Reply by hathach** [reply]
You can check out the above discussion, it may have some useful information to trouble shoot the issue. To be honest, it will be awhile until I could pull out 723 to test it.

**Comment by cdg66** [comment]
I have the same problem on my stm32f302 while debuging i say in _dcd_stm32_fsdev.c_ that isochronous transfer are not supported

>   // Isochronous not supported (yet), and some other driver assumptions.
  TU_ASSERT(p_endpoint_desc->bmAttributes.xfer != TUSB_XFER_ISOCHRONOUS);
  TU_ASSERT(epnum < MAX_EP_COUNT);

  // Set type
  switch(p_endpoint_desc->bmAttributes.xfer) {
  case TUSB_XFER_CONTROL:
    wType = USB_EP_CONTROL;
    break;
#if (0)
  case TUSB_XFER_ISOCHRONOUS: // FIXME: Not yet supported
    wType = USB_EP_ISOCHRONOUS;
    break;
#endif

but on an stm32F4 it work perfectly
